_pytest/fixtures.py

```diff
@@ -651,11 +661,14 @@ def formatrepr(self):
         if msg is None:
             fm = self.request._fixturemanager
             available = []
+            seen = set()
             parentid = self.request._pyfuncitem.parent.nodeid
             for name, fixturedefs in fm._arg2fixturedefs.items():
                 faclist = list(fm._matchfactories(fixturedefs, parentid))
-                if faclist:
+                name = strip_invocation_scope_suffix(name)
+                if faclist and name not in seen:
                     available.append(name)
+                    seen.add(name)
```

Comment:
Looks like ``available`` and ``seen`` hold the same data, both are only populated for unseen values. Maybe we can use just one? Technically if with the check up front a list would behave just like a set.